vnext/Shared/Networking/WinRTHttpResource.h

```diff
@@ -80,6 +83,12 @@ class WinRTHttpResource : public IHttpResource,
   void SetOnResponse(std::function<void(int64_t requestId, Response &&response)> &&handler) noexcept override;
   void SetOnData(std::function<void(int64_t requestId, std::string &&responseData)> &&handler) noexcept override;
   void SetOnData(std::function<void(int64_t requestId, folly::dynamic &&responseData)> &&handler) noexcept override;
+  void SetOnIncrementalData(
+      std::function<void(int64_t requestId, std::string &&responseData, int64_t progress, int64_t total)>
+          &&handler) noexcept override;
+  void SetOnDataProgress(
+      std::function<void(int64_t requestId, int64_t progress, int64_t total)> &&handler) noexcept override;
```

Comment:
>function [](http://example.com/codeflow?start=11&length=8)

(nit) can we specify that function that user passes should also be noexcept?

Reply:
Declaring those callbacks as `noexcept` crashes the component consistently.